src/test/regress/expected/background_rebalance_parallel.out

```diff
@@ -0,0 +1,279 @@
+
+(1 row)
+
+CREATE TABLE  table2_colg2 (b int primary key);
+SELECT create_distributed_table('table2_colg2', 'b' , colocate_with => 'table1_colg2');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+/* Colocation group 3: create two tables table1_colg3, table2_colg3 and in a colocation group  */
+CREATE TABLE table1_colg3 (a int PRIMARY KEY);
+SELECT create_distributed_table('table1_colg3 ', 'a', shard_count => 4, colocate_with => 'none');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE TABLE  table2_colg3 (b int primary key);
+SELECT create_distributed_table('table2_colg3', 'b' , colocate_with => 'table1_colg3');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+/* Add two new node so that we can rebalance */
+SELECT 1 FROM citus_add_node('localhost', :worker_3_port);
+ ?column?
+---------------------------------------------------------------------
+        1
+(1 row)
+
+SELECT 1 FROM citus_add_node('localhost', :worker_4_port);
+ ?column?
+---------------------------------------------------------------------
+        1
+(1 row)
+
+SELECT * FROM citus_rebalance_start();
+ citus_rebalance_start
+---------------------------------------------------------------------
+                 17777
+(1 row)
+
+SELECT citus_rebalance_wait();
+ citus_rebalance_wait
+---------------------------------------------------------------------
+
+(1 row)
+
+/*Check that a move is dependent on
+	1. any other move scheduled earlier in its colocation group.
+	2. any other move scheduled earlier whose source node or target
+	   node overlaps with the current moves nodes. */
+SELECT S.shardid, P.colocationid
+FROM pg_dist_shard S, pg_dist_partition P
+WHERE S.logicalrelid = P.logicalrelid ORDER BY S.shardid ASC;
+ shardid  | colocationid
+---------------------------------------------------------------------
+ 85674000 |      1390010
+ 85674001 |      1390010
+ 85674002 |      1390010
+ 85674003 |      1390010
+ 85674004 |      1390010
+ 85674005 |      1390010
+ 85674006 |      1390010
+ 85674007 |      1390010
+ 85674008 |      1390011
+ 85674009 |      1390011
+ 85674010 |      1390011
+ 85674011 |      1390011
+ 85674012 |      1390011
+ 85674013 |      1390011
+ 85674014 |      1390011
+ 85674015 |      1390011
+ 85674016 |      1390012
+ 85674017 |      1390012
+ 85674018 |      1390012
+ 85674019 |      1390012
+ 85674020 |      1390012
+ 85674021 |      1390012
+ 85674022 |      1390012
+ 85674023 |      1390012
+(24 rows)
+
+SELECT D.task_id,
+       (SELECT T.command FROM pg_dist_background_task T WHERE T.task_id = D.task_id),
+       D.depends_on,
+       (SELECT T.command FROM pg_dist_background_task T WHERE T.task_id = D.depends_on)
+FROM pg_dist_background_task_depend D  WHERE job_id = 17777 ORDER BY D.task_id ASC;
```

Comment:
```suggestion
FROM pg_dist_background_task_depend D  WHERE job_id = 17777 ORDER BY D.task_id, D.depends_on ASC;
```

Since task_id is not unique on its own

Reply:
same for the same query further down the file.